assets/user-profile/components/PauseNotificationModal.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {IUser} from 'interfaces';
+import {gettext} from 'utils';
+import {modalFormInvalid, modalFormValid} from 'actions';
+import {updateUserNotificationPause} from 'user-profile/actions';
+import Modal from 'components/Modal';
+
+interface IProps {
+    modalFormInvalid(): void;
+    modalFormValid(): void;
+    updateUserNotificationPause(schedule: Omit<IUser['notification_schedule'], 'last_run_time'>): void;
+    data: {
+        user: IUser;
+    };
+}
+
+interface IState {
+    pauseFrom?: string;
+    pauseTo?: string;
+}
+
+const disabledOptionDateFrom = new Date().toISOString().split('T')[0];
+
+class PauseNotificationModalComponent extends React.Component<IProps, IState> {
+    formRef: React.RefObject<HTMLFormElement>;
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            pauseFrom: this.props.data.user.notification_schedule?.pauseFrom ?? '',
+            pauseTo: this.props.data.user.notification_schedule?.pauseTo ?? '',
+        };
+
+        this.onSubmitForm = this.onSubmitForm.bind(this);
+        this.updateDate = this.updateDate.bind(this);
+        this.formRef = React.createRef<HTMLFormElement>();
+    }
+
+    componentDidMount() {
+        this.props.modalFormValid();
+    }
+
+    onSubmitForm(event: React.FormEvent<HTMLFormElement>) {
+        event.stopPropagation();
+        event.preventDefault();
+
+        if (this.formRef.current == null) {
+            throw new Error('ref missing');
+        }
+
```

Comment:
is this meant to stop the function execution? What does throwing an error serve us here?

Reply:
There are a few reasons for this. One is to stop the execution in case it is `null` and another is to tell compiler that if execution is not stopped, it should be treated as `non-null`.

We could also achieve this using `if/else` or early return. But - it is preferrable to throw errors when we're guarding against invalid application states.